src/TypeSpec.Extension/Emitter.Csharp/src/type/input-type.ts

```diff
@@ -1,34 +1,69 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+import { SdkBuiltInKinds } from "@azure-tools/typespec-client-generator-core";
 import { InputEnumTypeValue } from "./input-enum-type-value.js";
-import { InputIntrinsicTypeKind } from "./input-intrinsic-type-kind.js";
 import { InputModelProperty } from "./input-model-property.js";
-import { InputPrimitiveTypeKind } from "./input-primitive-type-kind.js";
 import { InputTypeKind } from "./input-type-kind.js";
+import {
+    DateTimeKnownEncoding,
+    DurationKnownEncoding
+} from "@typespec/compiler";
 
-export interface InputType {
-    Name: string;
-    Kind: InputTypeKind;
+interface InputTypeBase {
+    Kind: string;
```

Comment:
why is `Kind` string instead of `enum`?

Reply:
this is not public type.
In public types, kind is either an enum or a literal value.
I am just copying how it is implemented in TCGC: https://github.com/Azure/typespec-azure/blob/9bd912d514d625a5106cfd7c83228c7f86467587/packages/typespec-client-generator-core/src/interfaces.ts#L82
Currently we are not ready to just simply use their types, therefore I mimic their implementation here.
But in ts code, it is not a big deal, we just need to ensure the json schema is the same which is enough

Reply:
OK, since the contract is for public usage we should keep them `string`.
But when we assign the values, can we leverage `enum` to avoid typo? such as [this](https://github.com/Azure/autorest.csharp/blob/d2e20316508f3c013b4e88f9020f7e1476c7b088/src/TypeSpec.Extension/Emitter.Csharp/src/lib/converter.ts#L743)
And also, we are using const string for the mapping in generator [here](https://github.com/Azure/autorest.csharp/blob/d2e20316508f3c013b4e88f9020f7e1476c7b088/src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputTypeConverter.cs#L66-L73). Would it be possible to put them into an `enum` instead?

Since this is not introduced by the current PR, maybe a TODO would be good enough to be handled later.

Reply:
Yeah it will be changed to lower case in following PRs. Let me add to dos.

Reply:
TODOs added